lib/index.js

```diff
@@ -0,0 +1,186 @@
+        continue;
+      } else {
+        return false;
+      }
+    }
+
+    if (!/^:?-+:?$/.test(t)) { return false; }
+    if (t.charCodeAt(t.length - 1) === 0x3A/* : */) {
+      aligns.push(t.charCodeAt(0) === 0x3A/* : */ ? 'center' : 'right');
+    } else if (t.charCodeAt(0) === 0x3A/* : */) {
+      aligns.push('left');
+    } else {
+      aligns.push('');
+    }
+  }
+
+  lineText = getLine(state, startLine).trim();
+  if (lineText.indexOf('|') === -1) { return false; }
+  rows = escapedSplit(lineText.replace(/^\||\|$/g, ''));
+  if (aligns.length !== rows.length) { return false; }
+  if (silent) { return true; }
+
+  token     = state.push('table_open', 'table', 1);
+  token.map = tableLines = [ startLine, 0 ];
+
+  token     = state.push('thead_open', 'thead', 1);
+  token.map = [ startLine, startLine + 1 ];
+
+  token     = state.push('tr_open', 'tr', 1);
+  token.map = [ startLine, startLine + 1 ];
+
+  for (i = 0; i < rows.length; i++) {
+    token          = state.push('th_open', 'th', 1);
+    token.map      = [ startLine, startLine + 1 ];
+    if (aligns[i]) {
+      token.attrs  = [ [ 'align', aligns[i] ] ];
+    }
+
+    token          = state.push('inline', '', 0);
+    token.content  = rows[i].trim();
+    token.map      = [ startLine, startLine + 1 ];
+    token.children = [];
+
+    token          = state.push('th_close', 'th', -1);
+  }
+
+  token     = state.push('tr_close', 'tr', -1);
+  token     = state.push('thead_close', 'thead', -1);
+
+  token     = state.push('tbody_open', 'tbody', 1);
+  token.map = tbodyLines = [ startLine + 2, 0 ];
+
+  for (nextLine = startLine + 2; nextLine < endLine; nextLine++) {
+    if (state.sCount[nextLine] < state.blkIndent) { break; }
+
+    lineText = getLine(state, nextLine).trim();
+    if (lineText.indexOf('|') === -1) { break; }
+    rows = escapedSplit(lineText.replace(/^\||\|$/g, ''));
+
+    // set number of columns to number of columns in header row
+    rows.length = aligns.length;
+
+    token = state.push('tr_open', 'tr', 1);
+    for (i = 0; i < rows.length; i++) {
+      token          = state.push('td_open', 'td', 1);
+      if (aligns[i]) {
+        token.attrs  = [ [ 'align', aligns[i] ] ];
+      }
+
+      token          = state.push('inline', '', 0);
+      token.content  = rows[i] ? rows[i].trim() : '';
+      token.children = [];
+
+      token          = state.push('td_close', 'td', -1);
+    }
+    token = state.push('tr_close', 'tr', -1);
+  }
+  token = state.push('tbody_close', 'tbody', -1);
+  token = state.push('table_close', 'table', -1);
+
+  tableLines[1] = tbodyLines[1] = nextLine;
+  state.line = nextLine;
+  return true;
+};
+
+
+
+module.exports = function aligned_table_plugin(md, options) {
+  options = typeof options === 'object' ? options : {};
+
+  md.renderer.rules.aligned_table = md.renderer.rules.html_block;
+  md.block.ruler.before('table', 'aligned_table', table, {'alt': [ 'paragraph', 'reference' ]});
+
+}
```

Comment:
Stupid Idea: If all you've done was you basically searched&replaced `style="text-align: where"` for `align="where"`, could we treat this file basically as an artifact and make this library "generated"? i.e. it wouldn't contain `lib` files, there would be just command to build the lib (by downloading current code from upstream and doing the search&replace) and then it could be published as a package (similar way we publish js compiled from coffee sources).

Probably not worth at the moment, but in the future...?